### DISCUSSION TITLE
Use of rack-indifferent

### DISCUSSION THREAD

**Discussion by jeremyevans** [bridgetownrb/bridgetown#585] [General]
I currently maintain rack-indifferent, but I haven't used it in production in at least 5 years, and have thought using indifferent parameters is a bad idea for about as long.

Looking at reverse dependencies on rubygems, bridgetown-core is the only gem depending on rack-indifferent.  So instead of archiving the rack-indifferent repository (which I was planning to do), I'm fine transferring rack-indifferent to the bridgetownrb organization.  Let me know if you'd like me to do that.

Note that Roda itself offers an indifferent_params plugin that handles much the same need.  I'm not sure whether you've considered switching to that and dropping the rack-indifferent dependency.

**Comment by jaredcwhite** [comment]
@jeremyevans I can look into switching to `indifferent_params`…honestly I can't remember what caused me to use the extra dependency instead. I'd want to make sure we can still use `r.params` though in addition to simply `params`, otherwise it'd be a breaking change for user apps.

**Reply by jaredcwhite** [reply]
@jeremyevans I should have just run a quick test… yeah seems to be passing just fine. So I'd say go ahead and archive the repo, and I'll make sure to get the switch to the Roda plugin in for Bridgetown 1.1. Thanks for the heads up!

**Reply by jeremyevans** [reply]
Roda's indifferent_params plugin doesn't handle `r.params` (by design), so if you need that, you'll need some extra code on top of it, or to keep using rack-indifferent.  I think  from this comment that you are OK with that change, so I can archive rack-indifferent, but if not and you would me me to transfer it to you, I can do that.

**Reply by jaredcwhite** [reply]
@jeremyevans It looks to me like for Rack 2+, whether you call `params` or `r.params` you get the exact same indifferent behavior (in fact the `params` instance method really just aliases the request object's `params` method), so we're all good from my vantage point. 👍

**Reply by jeremyevans** [reply]
Yep, that's correct for Rack 2.  It's not for Rack 1 as the change would be too invasive.  So since Bridgetown can switch to the indifferent_params plugin, I'll archive rack-indifferent.
